check_NEON_data_products.R

```diff
@@ -1,31 +1,37 @@
+#install.packages("nneo")
 #Install packages needed to map
 #install.packages(c("ggplot2", "devtools", "dplyr", "stringr"))
 #install.packages(c("maps", "mapdata"))
+#install.packages("ggmap")
 
 library(ggplot2)
 library(ggmap)
 library(maps)
 library(mapdata)
 library(nneo)
 
+pdf("NEON_Site_Product_Availability.pdf", width=6, height=4)
 neon_sites <- nneo_sites()
+desiredProducts <- read.csv(file="NEON_Products.csv",head=TRUE)
 
 dta <-  map_data("world",c("usa","hawaii","alaska"),xlim=c(-180,-65), ylim=c(19,72))
 
-plot.new()
-gg1 <- ggplot() + geom_polygon(data = dta, aes(x=long, y=lat, group=group),fill=NA,colour="black") + coord_fixed(1.3)
-#gg1 <- gg1 + geom_point(data=neon_sites, aes(x = siteLongitude, y = siteLatitude)) #Can uncomment for all sites with data
+for (k in 1:(dim(desiredProducts)[1])){
```

Comment:
nrow(desiredProducts) is simpler than dim(desiredProducts)[1]

seq_len and seq_along are better practice for ```for``` loop indexing than 1:n because the latter behaves (and fails) in counterintuitive ways if n <= 0, while the former does what you expect (just skips the whole loop)

putting these together, my recommendation is seq_len(nrow(desiredProducts))